### ISSUE TITLE
npm install child process does not exit if output is captured

### ISSUE THREAD

**Issue by logiclrd** [sass/node-sass#2459]
I am encountering an odd issue. When the `node-sass` module is installed by `npm`, it launches a child process `node scripts/install.js`. I am finding that if the `npm install` parent process is launched with its standard streams redirected (to capture stdout and stderr -- I am using Ruby's `Open3::popen3`), then this `node scripts/install.js` process hangs and never exits. The last line in our build output looks like this:

    Cached binary found at C:\Users\Jonathan Gilbert\AppData\Roaming\npm-cache\node-sass\4.9.2\win32-x64-57_binding.node

If `npm install` is simply run from the command prompt, then it runs through to completion:

    C:\Users\Jonathan Gilbert\ProjectName>npm install
    
    > node-sass@4.9.2 install C:\Users\Jonathan Gilbert\ProjectName\node_modules\node-sass
    > node scripts/install.js
    
    Cached binary found at C:\Users\Jonathan Gilbert\AppData\Roaming\npm-cache\node-sass\4.9.2\win32-x64-57_binding.node
    
    > node-sass@4.9.2 postinstall C:\Users\Jonathan Gilbert\ProjectName\node_modules\node-sass
    > node scripts/build.js
    
    Binary found at C:\Users\Jonathan Gilbert\ProjectName\node_modules\node-sass\vendor\win32-x64-57\binding.node
    Testing binary
    Binary is fine
    ...

In practice what happens is that the `stdout` and `stderr` readers in the `popen3` block end up waiting forever because this child `node` process has inherited a handle to each pipe and never closes them -- so the streams never hit end-of-stream. I have found that I can instead wait for the `npm install` master process to exit, and then proceed ignoring the status of the `stdout` and `stderr` streams, but this has the problem that the `node scripts/install.js` process hangs around forever. Since this is Windows, it holds an open handle to the `./node_modules/node-sass` subdirectory, which blocks further tasks that want to alter this path.

I am at a loss for how to diagnose or solve this issue, and I am hoping that the developers in charge of the `node-sass` package itself may be able to understand what is going on here.

Version information:
- NPM version (`npm -v`): 6.1.0
- Node version (`node -v`): v8.1.3
- Node Process (`node -p process.versions`):

    { http_parser: '2.7.0',
      node: '8.1.3',
      v8: '5.8.283.41',
      uv: '1.12.0',
      zlib: '1.2.11',
      ares: '1.10.1-DEV',
      modules: '57',
      openssl: '1.0.2l',
      icu: '59.1',
      unicode: '9.0',
      cldr: '31.0.1',
      tz: '2017b' }

- Node Platform (`node -p process.platform`): win32
- Node architecture (`node -p process.arch`): x64
- node-sass version (`node -p "require('node-sass').info"`):

    node-sass       4.9.2   (Wrapper)       [JavaScript]
    libsass         3.5.4   (Sass Compiler) [C/C++]

- npm node-sass versions (`npm ls node-sass`): node-sass@4.9.2

**Comment by logiclrd** [comment]
A note to anybody trying to reproduce this: you don't need to write a `popen3` call to get this to happen, simply running `npm install` from Ruby using backticks to ask Ruby to capture the command output results in the same behaviour.
```
C:\code\Project>irb
irb(main):001:0> require 'fileutils'
irb(main):002:0> FileUtils.rm_rf("node_modules/node-sass") # force reinstallation
irb(main):003:0> `npm install`
█
```
Once it has reached the point of the hang, a process can be found with command-line `node scripts/install.js`, and I'm fairly certain this is referencing node-sass's `scripts/install.js`.

**Comment by xzyfer** [comment]
Nothing like this has been observed or reported in the 5+ years of this project, and I don't use Ruby. Honestly we don't know, but we're open to PRs if you manage to debug the issue.

**Comment by logiclrd** [comment]
I have done further debugging, and this appears to be a bug at the Node level. I haven't investigated and don't have the knowledge to simply know if this is true, but perhaps actually using install hook scripts is relatively rare, and I've observed this with `node-sass` because it's one of the few packages that actually uses one.

In my testing, I grabbed the source code for the same `node-sass` version, and changed my `popen3` test to use `npm install C:\Path\To\Local\Copy\Of\node-sass`. When I edit `scripts/install.js` to contain only a single `console.log` line (no actual logic), I still observe the hang. So, it isn't caused by anything `scripts/install.js` itself is doing.

In fact, if I outright delete `scripts/install.js`, then I get `Error: Cannot find module ...\scripts\install.js`, followed by many pages of warning messages, followed by some errors starting with `npm ERR: code ELIFECYCLE`, followed by the hang. Yes, it hangs _even if the install script isn't even present to run_.

Closing this issue and opening a bug on `node` instead. Thank you for your time!
